salt/metalk8s/volumes/prepared/files/metalk8s-volume-clean-up

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import os
+import subprocess
+import sys
+
+
+def die(message):
+    print(message, file=sys.stderr)
+    sys.exit(1)
+
+
+def get_volume_id():
+    if len(sys.argv) != 2:
+        die("Invalid number of arguments: expected only one volume ID")
+
+    return sys.argv[1]
+
+
+def find_device(volume_id):
+    path_by_uuid = "/dev/disk/by-uuid/{}".format(volume_id)
+    if os.path.exists(path_by_uuid):
+        return path_by_uuid
+
+    path_by_partuuid = "/dev/disk/by-partuuid/{}".format(volume_id)
+    if not os.path.exists(path_by_partuuid):
+        die("Volume '{}' backing device was not found".format(volume_id))
+
+    partition_name = os.path.basename(os.path.realpath(path_by_partuuid))
+    device_name = os.path.basename(
+        os.path.realpath("/sys/class/block/{}/..".format(partition_name))
+    )
+
+    return "/dev/{}".format(device_name)
+
+
+def clean_up(device_path):
+    try:
+        subprocess.check_output(
+            ["losetup", "-d", device_path],
```

Comment:
The original implementation in `salt/_modules/metalk8s_volumes.py` did the `losetup -d` in pure Python code. It could make sense to do the same instead of shelling out.

`losetup -d` implementation basically boils down to https://github.com/karelzak/util-linux/blob/a61955ba1f15bf0a252a03a9abe54c1e02828c2e/lib/loopdev.c#L1518

Reply:
Note that, *iff* we need to have a wrapper for the loop device creation (locking...) as well, we could consider to (re-)implement this in pure Python as well.

Reply:
Let's go for pure Python! Makes me think I have forgotten to remove the `losetup -d` from the `clean_up` method, in our module.

Reply:
Done (and removed the corresponding code from our custom module).